ayushma/views/chat.py

```diff
@@ -62,6 +63,13 @@ def get_queryset(self):
         return queryset
 
     def perform_create(self, serializer):
+        openai_key = self.request.headers.get("OpenAI-Key")
+
+        if (not openai_key and not serializer.validated_data.get("allow_key")):
```

Comment:
@rithviknishad the user cannot change the `allow_key` themselves. It can only be set by the admin